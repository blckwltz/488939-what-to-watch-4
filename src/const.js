export const TabNames = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};
