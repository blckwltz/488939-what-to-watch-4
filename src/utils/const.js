const BASE_URL = `https://4.react.pages.academy/wtw`;
const MAX_MOVIES_AMOUNT = 8;
const PLAYBACK_DELAY = 1000;
const TIMEOUT = 5;

const URL = {
  MOVIES: `/films`,
  FEATURED: `/films/promo`,
  FAVORITE: `/favorite`,
  REVIEWS: `/comments`,
  LOGIN: `/login`,
};

const Status = {
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  ERROR: 404,
  SERVER_ERROR: 500,
};

const TabNames = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const FilterSettings = {
  INITIAL_VALUE: `All genres`,
  MAX_AMOUNT: 9,
};

const Rating = {
  SCORE_MILESTONES: {
    FIRST: 0,
    SECOND: 3,
    THIRD: 5,
    FOURTH: 8,
    FIFTH: 10,
  },
  LEVELS: {
    BAD: `Bad`,
    NORMAL: `Normal`,
    GOOD: `Good`,
    VERY_GOOD: `Very Good`,
    AWESOME: `Awesome`,
  },
};

const RatingChart = new Map([
  [[Rating.SCORE_MILESTONES.FIRST, Rating.SCORE_MILESTONES.SECOND], Rating.LEVELS.BAD],
  [[Rating.SCORE_MILESTONES.SECOND, Rating.SCORE_MILESTONES.THIRD], Rating.LEVELS.NORMAL],
  [[Rating.SCORE_MILESTONES.THIRD, Rating.SCORE_MILESTONES.FOURTH], Rating.LEVELS.GOOD],
  [[Rating.SCORE_MILESTONES.FOURTH, Rating.SCORE_MILESTONES.FIFTH], Rating.LEVELS.VERY_GOOD],
  [[Rating.SCORE_MILESTONES.FIFTH, Infinity], Rating.LEVELS.AWESOME],
]);

export {BASE_URL, MAX_MOVIES_AMOUNT, PLAYBACK_DELAY, TIMEOUT, URL, Status, TabNames, FilterSettings, RatingChart};
