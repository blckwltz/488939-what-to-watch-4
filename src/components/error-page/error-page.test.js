import React from 'react';
import renderer from 'react-test-renderer';
import {Status} from '../../utils/const.js';
import ErrorPage from './error-page';

it(`Should render ErrorPage component correctly`, () => {
  const tree = renderer
    .create(
        <ErrorPage
          status={Status.BAD_REQUEST}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
