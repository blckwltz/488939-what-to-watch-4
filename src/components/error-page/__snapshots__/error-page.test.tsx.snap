// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ErrorPage component correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title visually-hidden"
    >
      Error
    </h1>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title"
    >
      Error 
      400
       occurred. Please try again later.
    </h2>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2020 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
